Getch doesn't work on windows

Here's the trace when trying to install the requirements via `pip`:

```shell
vksna@DESKTOP-RHUBCR5 MINGW64 ~/qh (main)
$ pip install -r requirements.txt
Collecting openai-whisper==20240930
  Using cached openai_whisper-20240930-py3-none-any.whl
Collecting google-api-python-client==2.93.0
  Using cached google_api_python_client-2.93.0-py2.py3-none-any.whl (11.4 MB)
Collecting google-auth==2.22.0
  Using cached google_auth-2.22.0-py2.py3-none-any.whl (181 kB)
Collecting google-auth-oauthlib==1.0.0
  Using cached google_auth_oauthlib-1.0.0-py2.py3-none-any.whl (18 kB)
Collecting google-auth-httplib2==0.1.0
  Using cached google_auth_httplib2-0.1.0-py2.py3-none-any.whl (9.3 kB)
Requirement already satisfied: pyperclip==1.8.2 in c:\users\vksna.000\appdata\local\programs\python\python39\lib\site-packages (from -r requirements.txt (line 6)) (1.8.2)
Collecting requests
  Using cached requests-2.32.3-py3-none-any.whl (64 kB)
Collecting mutagen
  Using cached mutagen-1.47.0-py3-none-any.whl (194 kB)
Collecting numpy
  Using cached numpy-2.0.2-cp39-cp39-win_amd64.whl (15.9 MB)
Collecting tqdm
  Using cached tqdm-4.67.1-py3-none-any.whl (78 kB)
Collecting getch
  Using cached getch-1.0.tar.gz (1.3 kB)
  Preparing metadata (setup.py) ... done
Collecting ffmpeg-python
  Using cached ffmpeg_python-0.2.0-py3-none-any.whl (25 kB)
Collecting more-itertools
  Using cached more_itertools-10.7.0-py3-none-any.whl (65 kB)
Collecting numba
  Using cached numba-0.60.0-cp39-cp39-win_amd64.whl (2.7 MB)
Collecting torch
  Using cached torch-2.7.0-cp39-cp39-win_amd64.whl (212.4 MB)
Collecting tiktoken
  Using cached tiktoken-0.9.0-cp39-cp39-win_amd64.whl (894 kB)
Collecting uritemplate<5,>=3.0.1
  Using cached uritemplate-4.1.1-py2.py3-none-any.whl (10 kB)
Collecting google-api-core!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.0,<3.0.0.dev0,>=1.31.5
  Using cached google_api_core-2.24.2-py3-none-any.whl (160 kB)
Collecting httplib2<1.dev0,>=0.15.0
  Using cached httplib2-0.22.0-py3-none-any.whl (96 kB)
Collecting cachetools<6.0,>=2.0.0
  Using cached cachetools-5.5.2-py3-none-any.whl (10 kB)
Collecting rsa<5,>=3.1.4
  Using cached rsa-4.9.1-py3-none-any.whl (34 kB)
Collecting six>=1.9.0
  Using cached six-1.17.0-py2.py3-none-any.whl (11 kB)
Collecting pyasn1-modules>=0.2.1
  Using cached pyasn1_modules-0.4.2-py3-none-any.whl (181 kB)
Collecting urllib3<2.0
  Using cached urllib3-1.26.20-py2.py3-none-any.whl (144 kB)
Collecting requests-oauthlib>=0.7.0
  Using cached requests_oauthlib-2.0.0-py2.py3-none-any.whl (24 kB)
Collecting certifi>=2017.4.17
  Using cached certifi-2025.4.26-py3-none-any.whl (159 kB)
Collecting charset-normalizer<4,>=2
  Using cached charset_normalizer-3.4.1-cp39-cp39-win_amd64.whl (102 kB)
Collecting idna<4,>=2.5
  Using cached idna-3.10-py3-none-any.whl (70 kB)
Collecting colorama
  Using cached colorama-0.4.6-py2.py3-none-any.whl (25 kB)
Collecting future
  Using cached future-1.0.0-py3-none-any.whl (491 kB)
Collecting googleapis-common-protos<2.0.0,>=1.56.2
  Using cached googleapis_common_protos-1.70.0-py3-none-any.whl (294 kB)
Collecting proto-plus<2.0.0,>=1.22.3
  Using cached proto_plus-1.26.1-py3-none-any.whl (50 kB)
Collecting protobuf!=3.20.0,!=3.20.1,!=4.21.0,!=4.21.1,!=4.21.2,!=4.21.3,!=4.21.4,!=4.21.5,<7.0.0,>=3.19.5
  Using cached protobuf-6.30.2-cp39-cp39-win_amd64.whl (431 kB)
Collecting pyparsing!=3.0.0,!=3.0.1,!=3.0.2,!=3.0.3,<4,>=2.4.2
  Using cached pyparsing-3.2.3-py3-none-any.whl (111 kB)
Collecting pyasn1<0.7.0,>=0.6.1
  Using cached pyasn1-0.6.1-py3-none-any.whl (83 kB)
Collecting oauthlib>=3.0.0
  Using cached oauthlib-3.2.2-py3-none-any.whl (151 kB)
Collecting llvmlite<0.44,>=0.43.0dev0
  Using cached llvmlite-0.43.0-cp39-cp39-win_amd64.whl (28.1 MB)
Collecting regex>=2022.1.18
  Using cached regex-2024.11.6-cp39-cp39-win_amd64.whl (274 kB)
Collecting jinja2
  Using cached jinja2-3.1.6-py3-none-any.whl (134 kB)
Collecting typing-extensions>=4.10.0
  Using cached typing_extensions-4.13.2-py3-none-any.whl (45 kB)
Collecting fsspec
  Using cached fsspec-2025.3.2-py3-none-any.whl (194 kB)
Collecting networkx
  Using cached networkx-3.2.1-py3-none-any.whl (1.6 MB)
Collecting filelock
  Using cached filelock-3.18.0-py3-none-any.whl (16 kB)
Collecting sympy>=1.13.3
  Using cached sympy-1.14.0-py3-none-any.whl (6.3 MB)
Requirement already satisfied: mpmath<1.4,>=1.1.0 in c:\users\vksna.000\appdata\local\programs\python\python39\lib\site-packages (from sympy>=1.13.3->torch->openai-whisper==20240930->-r requirements.txt (line 1)) (1.3.0)
Collecting MarkupSafe>=2.0
  Using cached MarkupSafe-3.0.2-cp39-cp39-win_amd64.whl (15 kB)
Using legacy 'setup.py install' for getch, since package 'wheel' is not installed.
Installing collected packages: getch, urllib3, uritemplate, typing-extensions, sympy, six, regex, pyparsing, pyasn1, protobuf, oauthlib, numpy, networkx, mutagen, more-itertools, MarkupSafe, llvmlite, idna, future, fsspec, filelock, colorama, charset-normalizer, certifi, cachetools, tqdm, rsa, requests, pyasn1-modules, proto-plus, numba, jinja2, httplib2, googleapis-common-protos, ffmpeg-python, torch, tiktoken, requests-oauthlib, google-auth, openai-whisper, google-auth-oauthlib, google-auth-httplib2, google-api-core, google-api-python-client
  Running setup.py install for getch ... error
  error: subprocess-exited-with-error

  × Running setup.py install for getch did not run successfully.
  │ exit code: 1
  ╰─> [11 lines of output]
      running install
      running build
      running build_ext
      building 'getch' extension
      creating build
      creating build\temp.win-amd64-3.9
      creating build\temp.win-amd64-3.9\Release
      C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.43.34808\bin\HostX86\x64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -I/usr/local/include -I/usr/include -IC:\Users\vksna.000\AppData\Local\Programs\Python\Python39\include -IC:\Users\vksna.000\AppData\Local\Programs\Python\Python39\include -IC:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.43.34808\include -IC:\Program Files\Microsoft Visual Studio\2022\Community\VC\Auxiliary\VS\include -IC:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\ucrt -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22000.0\\um -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22000.0\\shared -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22000.0\\winrt -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22000.0\\cppwinrt /Tcgetchmodule.c /Fobuild\temp.win-amd64-3.9\Release\getchmodule.obj
      getchmodule.c
      getchmodule.c(2): fatal error C1083: Cannot open include file: 'termios.h': No such file or directory
      error: command 'C:\\Program Files\\Microsoft Visual Studio\\2022\\Community\\VC\\Tools\\MSVC\\14.43.34808\\bin\\HostX86\\x64\\cl.exe' failed with exit code 2
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> getch

note: This is an issue with the package mentioned above, not pip.
hint: See above for output from the failure.
WARNING: You are using pip version 22.0.4; however, version 25.1 is available.
You should consider upgrading via the 'C:\Users\vksna.000\AppData\Local\Programs\Python\Python39\python.exe -m pip install --upgrade pip' command.
```

--- a/modules/labs.py
+++ b/modules/labs.py
@@ -4,13 +4,32 @@ import json
 import os
 import shutil
 import subprocess
-import sys
 import tempfile
-from getch import getch  # Requires `pip install getch`
+# from getch import getch
 
 import requests
 import whisper
 
+import sys
+
+if sys.platform.startswith('win'):
+    import msvcrt
+
+    def getch():
+        return msvcrt.getch().decode()
+else:
+    import tty
+    import termios
+
+    def getch():
+        fd = sys.stdin.fileno()
+        old_settings = termios.tcgetattr(fd)
+        try:
+            tty.setraw(fd)
+            ch = sys.stdin.read(1)
+        finally:
+            termios.tcsetattr(fd, termios.TCSADRAIN, old_settings)
+        return ch
 
 # --------------------------
 # Helper functions for transcript processing

--- a/requirements.txt
+++ b/requirements.txt
@@ -9,5 +9,5 @@ mutagen
 numpy
 tqdm
 # moviepy==1.0.3
-getch
+# getch
 ffmpeg-python

